aasemble/django/apps/mirrorsvc/models.py

```diff
@@ -163,3 +163,21 @@ def save(self, *args, **kwargs):
     def perform_snapshot(self):
         self.sync_dists()
         self.symlink_pool()
+
+    def user_can_modify(self, user):
+        return user == self.mirrorset.owner
+
+
+class Tags(models.Model):
+    snapshot = models.ForeignKey(Snapshot, related_name='tags')
+    tag = models.CharField(max_length=200)
+
+    class meta:
+        unique_together = ('snapshot', 'tag',)
+
+    def save(self, *args, **kwargs):
+        self.tag = self.tag.strip()
+        super(Tags, self).save(*args, **kwargs)
+
+    def user_can_modify(self, user):
+        return user == self.snapshot.mirrorset.owner
```

Comment:
I'd suggest instead invoking `self.snapshot.user_can_modify(user)`. That way, if we decide to change the logic for allowing edits, we'd only need to edit it in one place.